python/src/deltachat/direct_imap.py

```diff
@@ -159,9 +159,13 @@ def log(*args, **kwargs):
             log("---------", imapfolder, len(messages), "messages ---------")
             # get message content without auto-marking it as seen
             # fetching 'RFC822' would mark it as seen.
-            requested = [b'BODY.PEEK[HEADER]', FLAGS]
+            requested = [b'BODY.PEEK[]', FLAGS]
             for uid, data in self.conn.fetch(messages, requested).items():
-                body_bytes = data[b'BODY[HEADER]']
+                body_bytes = data[b'BODY[]']
+                if not body_bytes:
+                    log("Message", uid, "has empty body")
+                    continue
+
```

Comment:
FYI on test failures pytest prints a complete dump of remote INBOX and DeltaChat folders -- this fix here makes sure the full mime message is dumped properly. These will sometimes be encrypted, though.  If you need inspection you can call `set_config("e2ee_enabled", "0")` on the respective account in the respective test, and then you will get all cleartext dumps. Ideally we would have an easy FFI to decrypt a mime message in full.